dev/WebView2/InteractionTests/WebView2Tests.cs

```diff
@@ -0,0 +1,2932 @@
+        {
+            var result = new Edit(FindElement.ById("TestResult"));
+            var check = new CheckBox(FindElement.ById("MessageReceived"));
+            Button CompleteTestButton = new Button(FindElement.ById("CompleteTest"));
+            string expectedValue = selectedTest + ": Passed []";
+
+            // If the web message hasn't been received, wait for it
+            using (var waiter = check.GetToggledWaiter())
+            {
+                Log.Comment("Waiting for result...");
+                CompleteTestButton.Invoke();
+
+                if (check.ToggleState != ToggleState.On && !waiter.TryWait(TimeSpan.FromSeconds(5)))
+                {
+                    Log.Error("Expected web message was never received.");
+                }
+            }
+
+            // By now, we'll have gotten a pass/fail result
+            Verify.AreEqual(expectedValue, result.Value);
+        }
+
+        private static void CompleteTestAndWaitForResult(string selectedTest)
+        {
+            var result = new Edit(FindElement.ById("TestResult"));
+            Button CompleteTestButton = new Button(FindElement.ById("CompleteTest"));
+            string expectedValue = selectedTest + ": Passed []";
+
+            Log.Comment("Waiting for result...");
+            using (var waiter = new ValueChangedEventWaiter(result, expectedValue))
+            {
+                CompleteTestButton.Invoke();
+                waiter.TryWait();
+            }
+
+            for (int i = 0; i <= 2; i++)
+            {
+                if (expectedValue != result.Value)
+                {
+                    Log.Comment($"CompleteTestAndWaitForResult: Expected '{expectedValue}', result was '{result.Value}'. Waiting...");
+                    Thread.Sleep(TimeSpan.FromMilliseconds(500));
+                }
+            }
+
+            Verify.AreEqual(expectedValue, result.Value);
+        }
+
+        private static void WaitForTextBoxValue(Edit textbox, string expectedValue, bool match = true)
+        {
+            using (var waiter = new ValueChangedEventWaiter(textbox, expectedValue))
+            {
+                // if value is empty, wait for it
+                if (textbox.Value == string.Empty)
+                {
+                    try
+                    {
+                        waiter.Wait(TimeSpan.FromSeconds(120));
+                    }
+                    catch
+                    {
+                        Log.Comment("Wait timed out, textbox value is {0}", textbox.Value);
+                    }
+                }
+            }
+
+            if (match == false)
+            {
+                if (textbox.Value == expectedValue)
+                {
+                    return;
+                }
+            }
+            else
+            {
+                if (textbox.Value != expectedValue)
+                {
+                    return;
```

Comment:
shouldn't this assert/throw/fail the test?

Reply:
Hmmm .. I don't know. This is what it looks like in WinUI3's code as well. @dmitriykomin ?

Reply:
Good catch. This function does two things (1)wait for textbox to attain a new value, and then (2) match that against expectedString (or the negation of it). 

In the two places we use it, looks like it *mostly* just depends on part (1) - if the textbox value, changes then things are good, we don't depend on (2)... I am not sure that's true 100% of the time it's called, and also there could be new usages in the future expecting the matching part to work correctly...

In short, I agree this should be changed to return a bool reflecting the matching result.